src/objective-c/GRPCClient/private/GRPCHost.m

```diff
@@ -113,6 +113,12 @@ + (void)flushChannelCache {
   }
 }
 
++ (void)clearAllHostsForTesting {
```

Comment:
What about `resetHostSettings`? (It's not obvious to the user that there's a cache involved, as they just setting configs for specific hosts. I might not even be right from us to call it a cache in the implementation, as it's not really a trick to improve performance but rather a store).